central/metrics/aggregator/common/common.go

```diff
@@ -0,0 +1,75 @@
+package common
+
+import (
+	"context"
+	"iter"
+	"regexp"
+	"slices"
+
+	"github.com/pkg/errors"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	log = logging.CreateLogger(logging.ModuleForName("central_metrics"), 1)
+
+	// Source: https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+	metricNamePattern = regexp.MustCompile("^[a-zA-Z_:][a-zA-Z0-9_:]*$")
+)
+
+type Label string      // Prometheus label.
+type MetricName string // Prometheus metric name.
+type Countable interface{ Count() int }
+type FindingGenerator[Finding Countable] func(context.Context, *v1.Query, MetricsConfiguration) iter.Seq[Finding]
+type LabelGetter[Finding Countable] struct {
+	Label  Label
+	Getter func(Finding) string
+}
+
+// MetricsConfiguration is the parsed aggregation configuration.
+type MetricsConfiguration map[MetricName]map[Label]Expression
+
+func (mcfg MetricsConfiguration) HasAnyLabelOf(labels []Label) bool {
+	for _, labelExpr := range mcfg {
+		for label := range labelExpr {
+			if slices.Contains(labels, label) {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
This smells like we could use some optimization from  https://pkg.go.dev/maps. Could we somehow smartly apply the `maps.Values` here? (optional)